fl4health/server/model_merge_server.py

```diff
@@ -0,0 +1,104 @@
+import datetime
+import timeit
+from logging import DEBUG, INFO
+from typing import Optional, Tuple
+
+from flwr.common.logger import log
+from flwr.server.client_manager import ClientManager
+from flwr.server.history import History
+from flwr.server.server import Server
+from flwr.server.strategy import Strategy
+
+from fl4health.reporting.metrics import MetricsReporter
+from fl4health.strategies.model_merge_strategy import ModelMergeStrategy
+
+
+class ModelMergeServer(Server):
+    def __init__(
+        self,
+        *,
+        client_manager: ClientManager,
+        strategy: Optional[Strategy] = None,
+        metrics_reporter: Optional[MetricsReporter] = None,
+    ) -> None:
+        """
+        ModelMergeServer provides functionality to fetch client weights, perform a simple average,
+            redistirbute to clients for evaluation. Optionally can perform server side evaluation as well.
+        Args:
+            client_manager (ClientManager): Determines the mechanism by which clients are sampled by the server, if
+                they are to be sampled at all.
+            strategy (Optional[Strategy], optional): The aggregation strategy to be used by the server to handle.
+                client updates and other information potentially sent by the participating clients. If None the
+                strategy is FedAvg as set by the flwr Server.
+            metrics_reporter (Optional[MetricsReporter], optional): A metrics reporter instance to record the metrics
+                during the execution. Defaults to an instance of MetricsReporter with default init parameters.
+        """
+        assert isinstance(strategy, ModelMergeStrategy)
+        super().__init__(client_manager=client_manager, strategy=strategy)
+
+        if metrics_reporter is not None:
+            self.metrics_reporter = metrics_reporter
+        else:
+            self.metrics_reporter = MetricsReporter()
+
+    def fit(self, num_rounds: int, timeout: Optional[float]) -> Tuple[History, float]:
+        """
+        Performs a fit round in which the local client weights are evaluated on their test set,
+            uploaded to the server and averaged, then redistributed to clients for evaluation.
+            Optionally, can perform evaluation of the merged model on the server side as well.
+
+        Args:
+            num_rounds (int): Not used.
+            timeout (Optional[float]): Timeout in seconds that the server should wait for the clients to response.
```

Comment:
Maybe this was a typo that you copied from somewhere else but I think we want "respond" here 🙂

Reply:
Yup, will fix this in the other place too!